lte/cloud/go/services/subscriberdb_cache/worker.go

```diff
@@ -66,41 +69,120 @@ func RenewDigests(
 	if err != nil {
 		return nil, nil, errors.Wrapf(err, "remove per sub digests of invalid networks")
 	}
+	err = subStore.DeleteSubscribersForNetworks(networksToRemove)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "remove sub protos of invalid networks")
+	}
 
 	errs := &multierror.Error{}
-	flatDigestsByNetwork := map[string]string{}
 	perSubDigestsByNetwork := map[string][]*lte_protos.SubscriberDigestWithID{}
+	flatDigestsByNetwork := map[string]string{}
 	for _, network := range networksToRenew {
-		// The per-sub digests in store are updated en masse (collectively serialized into one blob per network);
-		// this update takes place along with every flat digest update for consistency.
-		// If an error occurs during this step, the overall last_updated_at timestamp for the network will
-		// not update, and will indicate outdated-ness instead, forcing a redo in the next loop.
-		perSubDigests, err := subscriberdb.GetPerSubscriberDigests(network)
+		digest, perSubDigests, err := renewDigestsForNetwork(network, digestStore, perSubDigestStore, subStore)
 		if err != nil {
-			multierror.Append(errors.Wrapf(err, "get per sub dgests to update"))
-			continue
+			multierror.Append(errs, err)
 		}
-		err = perSubDigestStore.SetDigest(network, perSubDigests)
+		flatDigestsByNetwork[network] = digest
+		perSubDigestsByNetwork[network] = perSubDigests
+	}
+
+	return flatDigestsByNetwork, perSubDigestsByNetwork, errs.ErrorOrNil()
+}
+
+// renewDigestsForNetwork updates the digest stores and subscriber proto cache for a given network.
+func renewDigestsForNetwork(
+	network string,
+	digestStore storage.DigestStore,
+	perSubDigestStore *storage.PerSubDigestStore,
+	subStore *storage.SubStore,
+) (string, []*lte_protos.SubscriberDigestWithID, error) {
+	errs := &multierror.Error{}
+	digest, prevDigest, err := updateDigest(network, digestStore)
+	if err != nil {
+		multierror.Append(errs, err)
+	}
+
+	var perSubDigests []*lte_protos.SubscriberDigestWithID
+	if errs.ErrorOrNil() == nil {
```

Comment:
`errs.ErrorOrNil() == nil` is an idiosyncratic construction. Just use named single errs (e.g. `updateDigestErr`, then e.g. before your final return you can aggregate them into a multi-error